sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveShim.scala

```diff
@@ -1140,6 +1140,44 @@ private[client] class Shim_v0_13 extends Shim_v0_12 {
     partitions.asScala.toSeq
   }
 
+  private def getPartitionNamesWithCount(hive: Hive, table: Table): (Int, Seq[String]) = {
+    val partitionNames = hive.getPartitionNames(
+      table.getDbName, table.getTableName, -1).asScala.toSeq
+    (partitionNames.length, partitionNames)
+  }
+
+  private def getPartitionsInBatches(
+      hive: Hive,
+      table: Table,
+      size: Int,
+      partNames: Seq[String]): java.util.Collection[Partition] = {
+    logInfo(s"Starting the batch processing of partitions.")
+
+    val retryCount = SQLConf.get.metastorePartitionBatchRetryCount
+    val batches = partNames.grouped(size).toSeq
+    val processedPartitions = batches.flatMap { batch =>
+      var partitions: java.util.Collection[Partition] = null
+      var retry = 0
+      while (partitions == null && retry < retryCount) {
+        try {
+          partitions = hive.getPartitionsByNames(table, batch.asJava)
+        } catch {
+          case ex: Exception =>
+            logWarning(
+              s"Caught exception while fetching partition metadata for batch '$batch'.",
+              ex
+            )
+            retry += 1
+            if (retry > retryCount) {
```

Comment:
I don't get the point of involving a retry mechanism here. In HIVE-27505, a BATCH_DECAYING_FACTOR is introduced to decrease the batch size after encountering an exception and trying again. Here we use a fixed batch size and retry makes no sense

Reply:
I thought if the current getPartitionbyNames fails due to connectivity or some issue, it should retry to fetch same partition again. Sure I will try to look into the decaying factor parameter as well.